runtime/src/main/java/org/corfudb/runtime/view/LayoutManagementView.java

```diff
@@ -394,6 +406,10 @@ private void attemptConsensus(Layout layout)
             }
             log.warn("attemptConsensus: Checking if {} has been accepted, retry {}", layout, x);
         }
+
+        // If our proposed layout was not selected, throw an exception.
+        throw new IllegalStateException(String.format("Layout %s was not accepted. " +
```

Comment:
This throw of exception actually fixes the flaky tests we have, in particular the ones where the workflows are happening in parallel (ConcurrentCluserResizeIT). One of the common correctness issues that I witnessed before this fix was introduced was in the Heal or Add workflow steps. For example, 5 nodes are being added at the same time, and each one of them propose a layout after (AddNodeToLayout) action. The node think that it's layout was proposed (since we don't through this exception) but it was not. As a result, a node performs the next step (StateTransfer) based on the incorrect layout, and this StateTransfer never succeeds.

Reply:
Was removed, needs discussion and a possible fix in some other PR.